### PR TITLE
reference --> dependency

### PR DESCRIPTION
**Pull request by elharo**
@suztomo

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ permalink: /glossary
 - **Linkage error**: an abnormal condition of a classpath in which a
   "class has some dependency on another class; however, the
   latter class has incompatibly changed after the compilation of the
-  former class."<sup>[1](#myfootnote1)</sup> The reference can be
+  former class."<sup>[1](#myfootnote1)</sup> The dependency can be
   through a class literal, a field access, or a method invocation.
```
Comment (suztomo) [inline docs/glossary.md:28]:
```suggestion
  symbolic reference to a class literal, a field access, or a method invocation.
```

Comment (elharo) [reply docs/glossary.md:28]: That introduces a new term here that is not defined and not otherwise used on this page.

Comment (suztomo) [reply docs/glossary.md:28]: @elharo I thought about the difference between "dependency" and "symbolic reference". The symbolic references in the constant pool of class files are the implementation detail of the "dependency" written in the [java.lang.LinkageError Javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/LinkageError.html).  Therefore the term "symbolic reference" does not belong to this doc.

Does this match your thought?

Comment (elharo) [reply docs/glossary.md:28]: It's simpler than that. We're quoting here from the official JDK docs, with citation. We use the same term they use. We do not need to introduce new and unfamiliar terms to define a linkage error here.

Comment (suztomo) [reply docs/glossary.md:28]: Okay.

**Pull request merged**